lib/database-ops.js

```diff
@@ -101,46 +101,64 @@ var changeUserPasswordInDatabase = function(userID, hashedPassword) {
-            return;
-        }
+    MongoClient.connect(dbURL).then((db) => {
         logger.info("Successfully connected to database.");
         database = db;
 
-        var finishInitialization = function () {
-            imageEntryCollection = database.collection("image-entries");
+        var finishInitialization = function () {  
             commentsCollection = database.collection("comments");
             logCollection = database.collection("logs");
             actionHistoryCollection = database.collection("action-history");
 
             usersCollection.createIndex({ username: 1 }, { unique: true, sparse: true });
-            callback(err);
+            callback();
         };
 
-        database.listCollections().toArray(function(err, collections) {
-            if (err) {
-                throw err;
+        var CheckForImageEntriesCollection = function(collectionNames) {
+            if(collectionNames.indexOf("image-entries") < 0) {
+                database.createCollection("image-entries").then((collection) => {
+                    collection.createIndex({ uploadeddate: 1 }, { 
+                        name: "expire_index",
+                        expireAfterSeconds: 300,
+                        partialFilterExpression: {
+                            temp: {
+                                $eq: true,
+                            }
+                        } 
+                    });
+                    imageEntryCollection = collection;
+                    logger.info("Images collection created.");
+                    finishInitialization();
+                });
+            }
+            else {
+                imageEntryCollection = database.collection("image-entries");
+                finishInitialization();
             }
+        }
 
+        database.listCollections().toArray().then((collections) => {
             var collectionNames = collections.map(function(collection) {
                 return collection.name;
             });
+            
             if (collectionNames.indexOf("users") < 0) {
-                database.createCollection("users", {collation: {locale: "en", strength: 2}}, function(err, collection) {
-                    if (err) {
-                        throw err;
-                    }
+                database.createCollection("users", {collation: {locale: "en", strength: 2}}).then((collection) => {
                     usersCollection = collection;
                     logger.info("Users collection created.");
-                    finishInitialization();
+                    CheckForImageEntriesCollection(collectionNames);
+                }).catch((err) => {
+                    throw err;
                 });
             } else {
                 usersCollection = database.collection("users");
-                finishInitialization();
+                CheckForImageEntriesCollection(collectionNames);
             }
+        }).catch((err) => {
+            throw err;
```

Comment:
```suggestion
            callback(err);
```

The `catch` below will not catch error if thrown here. `UnhandledPromiseRejectionWarning` will happen instead. Better to call the callback in this particular case.